source/DSCResources/DSC_CMForestDiscovery/DSC_CMForestDiscovery.psm1

```diff
@@ -128,38 +140,46 @@ function Set-TargetResource
                 }
             }
 
-            if (-not [string]::IsNullOrEmpty($PollingSchedule))
+            if ($PSBoundParameters.ContainsKey('ScheduleInterval') -and $PSBoundParameters.ContainsKey('ScheduleCount'))
             {
-                $newSchedule = @{
-                    RecurInterval = $PollingSchedule.RecurInterval
-                    RecurCount    = $PollingSchedule.RecurCount
+                if ($ScheduleInterval -eq 'Days' -and $ScheduleCount -ge 32)
+                {
+                    Write-Warning -Message ($script:localizedData.MaxIntervalDays -f $ScheduleCount)
+                    $scheduleCheck = 31
                 }
-
-                $desiredPollingSchedule = New-CMSchedule @newSchedule
-
-                $currentSchedule = @{
-                    RecurInterval = $state.PollingSchedule.RecurInterval
-                    RecurCount    = $state.PollingSchedule.RecurCount
+                elseif ($ScheduleInterval -eq 'Hours' -and $ScheduleCount -ge 24)
+                {
+                    Write-Warning -Message ($script:localizedData.MaxIntervalHours -f $ScheduleCount)
+                    $scheduleCheck = 23
+                }
+                else
+                {
+                    $scheduleCheck = $ScheduleCount
                 }
 
-                $stateSchedule = New-CMSchedule @currentSchedule
-
-                $array = @('DayDuration','DaySpan','HourDuration','HourSpan','IsGMT')
+                if ($ScheduleInterval -ne $state.ScheduleInterval)
+                {
+                    Write-Verbose -Message ($script:localizedData.SIntervalSet -f $ScheduleInterval)
+                    $setSchedule = $true
+                }
 
-                foreach ($item in $array)
+                if ($scheduleCheck -ne $state.ScheduleCount)
                 {
-                    if ($desiredPollingSchedule.$($item) -ne $stateSchedule.$($item))
-                    {
-                        Write-Verbose -Message ($script:localizedData.ScheduleItem `
-                            -f $item, $($desiredPollingSchedule.$($item)), $($stateSchedule.$($item)))
-                        $setSchedule = $true
-                    }
+                    Write-Verbose -Message ($script:localizedData.SCountSet -f $ScheduleCount)
+                    $setSchedule = $true
                 }
 
-                if ($setSchedule)
+                if ($setSchedule -eq $true)
                 {
+                    $pScheduleSet = @{
+                        RecurInterval = $ScheduleInterval
+                        RecurCount    = $ScheduleCount
```

Comment:
RecurCount = $ScheduleCheck. Currently works because test is looking at scheduleCheck and the underlying commandlets have nice behavior when the max is exceeded, but we should set this to the right variable.